upath/core.py

```diff
@@ -376,15 +388,30 @@ def glob(self: PT, pattern: str) -> Generator[PT, None, None]:
             name = name.split(self._flavour.sep)
             yield self._make_child(name)
 
-    def rglob(self: PT, pattern: str) -> Generator[PT, None, None]:
-        path_pattern = self.joinpath(pattern)
-        r_path_pattern = self.joinpath("**", pattern)
-        for p in (path_pattern, r_path_pattern):
-            for name in self._accessor.glob(self, p):
+    if _FSSPEC_HAS_WORKING_GLOB:
```

Comment:
Would it make sense to perform this check lazily on first use of `rglob`?

Reply:
Yes. it'll make this probably easier to read.
I'll make the change in the next days.

We should also wait for fsspec/adlfs#422 to fix the failing azure tests